tests/actions/ReportFieldsTest.ts

```diff
@@ -0,0 +1,237 @@
+        });
+
+        it('creates a new date report field of a workspace', async () => {
+            mockFetch?.pause?.();
+            Onyx.set(ONYXKEYS.FORMS.WORKSPACE_REPORT_FIELDS_FORM_DRAFT, {});
+            await waitForBatchedUpdates();
+
+            const policyID = Policy.generatePolicyID();
+            const reportFieldName = 'Test Field 2';
+            const reportFieldID = generateFieldID(reportFieldName);
+            const reportFieldKey = ReportUtils.getReportFieldKey(reportFieldID);
+            const defaultDate = DateUtils.extractDate(new Date().toString());
+            const newReportField: Omit<PolicyReportField, 'value'> = {
+                name: reportFieldName,
+                type: CONST.REPORT_FIELD_TYPES.DATE,
+                defaultValue: defaultDate,
+                values: [],
+                disabledOptions: [],
+                fieldID: reportFieldID,
+                orderWeight: 1,
+                deletable: false,
+                keys: [],
+                externalIDs: [],
+                isTax: false,
+            };
+            const createReportFieldArguments: CreateReportFieldArguments = {
+                name: reportFieldName,
+                type: CONST.REPORT_FIELD_TYPES.DATE,
+                initialValue: defaultDate,
+            };
+
+            ReportFields.createReportField(policyID, createReportFieldArguments);
+            await waitForBatchedUpdates();
+
+            let policy: OnyxEntry<PolicyType> | OnyxCollection<PolicyType> = await new Promise((resolve) => {
+                const connectionID = Onyx.connect({
+                    key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+                    callback: (workspace) => {
+                        Onyx.disconnect(connectionID);
+                        resolve(workspace);
+                    },
+                });
+            });
+
+            // check if the new report field was added to the policy
+            expect(policy?.fieldList).toStrictEqual({
+                [reportFieldKey]: newReportField,
+            });
+
+            // Check for success data
+            (fetch as MockFetch)?.resume?.();
+            await waitForBatchedUpdates();
+
+            policy = await new Promise((resolve) => {
+                const connectionID = Onyx.connect({
+                    key: ONYXKEYS.COLLECTION.POLICY,
+                    waitForCollectionCallback: true,
+                    callback: (workspace) => {
+                        Onyx.disconnect(connectionID);
+                        resolve(workspace);
+                    },
+                });
+            });
+
+            // Check if the policy pending action was cleared
+            // @ts-expect-error pendingFields is not null
+            expect(policy?.pendingFields?.[reportFieldKey]).toBeFalsy();
+        });
+
+        it('creates a new list report field of a workspace', async () => {
```

Comment:
@rezkiy37 can you please add Edit and delete flows to the unit tests in a follow up PR? Thanks

Reply:
Yes, I will do it.